src/driver.rs

```diff
@@ -120,6 +122,40 @@ pub fn main() {
                 exit(0);
             }
 
+            if std::env::args().any(|a| a == "--help" || a == "-h") {
+                println!(
+                    "\
+Checks a package to catch common mistakes and improve your Rust code.
+
+Usage:
+    cargo clippy [options] [--] [<opts>...]
+
+Common options:
+    -h, --help               Print this message
+    -V, --version            Print version info and exit
+
+Other options are the same as `cargo check`.
+
+To allow or deny a lint from the command line you can use `cargo clippy --`
+with:
+
+    -W --warn OPT       Set lint warnings
+    -A --allow OPT      Set lint allowed
+    -D --deny OPT       Set lint denied
+    -F --forbid OPT     Set lint forbidden
+
+You can use tool lints to allow or deny lints from your code, eg.:
+
+    #[allow(clippy::needless_lifetimes)]
+"
+                );
+
+                for lint in &lintlist::ALL_LINTS[..] {
+                    println!("clippy::{},", lint.name);
```

Comment:
We should only list these if you type `-W help`, (also A, D, F). See `rustc --help` vs `rustc -W help`

We should list the lint groups here though.

Reply:
yea this part wasn't meant to stay, its just to show its working and serve as a jumping off point for discussion.